modules/ppcp-button/src/Assets/SmartButton.php

```diff
@@ -1318,8 +1318,22 @@ public function script_data(): array {
 			'should_handle_shipping_in_paypal'        => $this->should_handle_shipping_in_paypal && ! $this->is_checkout(),
 			'needShipping'                            => $this->need_shipping(),
 			'vaultingEnabled'                         => $this->settings->has( 'vault_enabled' ) && $this->settings->get( 'vault_enabled' ),
+			'productType'                             => null,
+			'manualRenewalEnabled'                    => '',
 		);
 
+		if ( is_product() ) {
+			$product = wc_get_product( get_the_ID() );
+			if ( is_a( $product, \WC_Product::class ) ) {
+				$localize['productType'] = $product->get_type();
+			}
+		}
+
+		if ( class_exists( '\WCS_Manual_Renewal_Manager' ) ) {
+			/** @psalm-suppress UndefinedClass */
```

Comment:
Also there is the annoying WP phpcs error about missing description :)

Another way could be to add the class stub into https://github.com/woocommerce/woocommerce-paypal-payments/blob/trunk/.psalm/wcs.php